packages/protocol/src/mdns/MdnsClient.ts

```diff
@@ -1528,32 +1528,54 @@ export class MdnsClient implements Scanner {
         }
     }
 
-    static discoveryDataDiagnostics(data: DiscoveryData) {
-        return Diagnostic.dict(
-            {
-                SII: data.SII,
-                SAI: data.SAI,
-                SAT: data.SAT,
-                T: data.T,
-                DT: data.DT,
-                PH: data.PH,
-                ICD: data.ICD,
-                VP: data.VP,
-                DN: data.DN,
-                RI: data.RI,
-                PI: data.PI,
-            },
-            true,
-        );
+    static discoveryDataDiagnostics(data: DiscoveryData, kind?: string) {
+        return Diagnostic.dict({
+            kind,
+            SII: data.SII && Interval.format(data.SII),
```

Comment:
Allow format to get undefined and return that if passed, so you bypass such safe guards?

Reply:
Yeah this is kind of a limited use but I guess that's OK